crypto/err/err.c

```diff
@@ -213,9 +216,13 @@ static void build_SYS_str_reasons(void)
 
         str->error = ERR_PACK(ERR_LIB_SYS, 0, i);
         if (str->string == NULL) {
-            char (*dest)[LEN_SYS_STR_REASON] = &(strerror_tab[i - 1]);
-            if (openssl_strerror_r(i, *dest, sizeof(*dest)))
-                str->string = *dest;
+            if (openssl_strerror_r(i, cur, sizeof(strerror_pool - cnt))) {
+                size_t l = strlen(cur) + 1;
+
+                str->string = cur;
+                cnt += l;
```

Comment:
Shouldn't there be an overflow check here?

Reply:
Fixed